Lib/test/test_pdb.py

```diff
@@ -1123,9 +1130,136 @@ def tearDown(self):
         support.unlink(support.TESTFN)
 
 
+class PdbModuleTestCase(PdbBaseTestCase):
+    """Re-runs all tests used for a script but using a module"""
```

Comment:
Our preferred structure for this kind of testing is to put the actual test methods in a base class that *isn't* a `unittest.TestCase` subclass (so we only write them once and structurally ensure that they're consistent across test scenarios), and then add `unittest.TestCase` as a mixin in the subclasses.

Having two copies of the test methods is significantly less desirable, as that makes it easier for the test cases to diverge over time.

Reply:
There is actually not that much being shared between the two test cases other than the "run module".
I guess you were referring to something like [this](https://github.com/python/cpython/blob/6f0eb93183519024cb360162bdd81b9faec97ba6/Lib/test/test_iterlen.py#L84)

When I was going through the work I realized there is not that many tests in that test case that made sense running with both module/script. Most "basic" unit tests are covered in the doctests. If you want I can try to rework the whole suite (moving the doctests?) and the do the mixing thing. otherwise I've added a commit that just adds the new helper function to run a module and left the tests in the same suite.